### ISSUE TITLE
failures with unit period

### ISSUE THREAD

**Issue by rveltz** [RalphAS/PeriodicSchurDecompositions.jl#9]
Hi,

I think there is a corner case that is missing

```julia
partial_pschur([rand(100,100)+I],10) #errors
partial_pschur([rand(100,100)+I,rand(100,100)+I],10) #does not
```

**Comment by RalphAS** [comment]
Your case results (mainly?) from type instability of some generator expressions, which cause trouble for several functions in this package. (So I'm generalizing the issue title.) I'll try to be systematic about correcting for this (and adding tests).

**Comment by rveltz** [comment]
basically, I am using your package for shooting, hence I might as well need to have a single shooting. I dont want to call a different function depending on the number of shooting planes

**Comment by RalphAS** [comment]
That's reasonable.  I have fixed (on a local branch) the crash-causing indexing and type issues for unit period, but some of the algorithms needed for Krylov-Schur are not yet working correctly for that case, so I'll keep working on it.

**Comment by RalphAS** [comment]
This issue should be fixed in v0.1.2
